x-pack/plugins/alerting/server/application/rule/transforms/transform_raw_actions_to_domain_actions.ts

```diff
@@ -42,8 +41,8 @@ export const transformRawActionsToDomainActions = ({
         uuid: action.uuid,
         ...(action.frequency ? { frequency: action.frequency } : {}),
         ...(action.alertsFilter ? { alertsFilter: action.alertsFilter } : {}),
-        ...(action.useAlertDataAsTemplate
-          ? { useAlertDataAsTemplate: action.useAlertDataAsTemplate }
+        ...(action.useAlertDataForTemplate
+          ? { useAlertDataForTemplate: action.useAlertDataForTemplate }
```

Comment:
I guess this field (`useAlertDataAsTemplate`) was overlooked when it was renamed.

Reply:
Did we validate this with @cnasikas ?  I think had done some blame research, looked like it was a PR from him?

Reply:
Christos has already ack'ed it.
I did this change to unblock the PR, and I think this is the correct solution.

Reply:
Sorry, the ack was for the message in Slack (that I have read and I will take a look at soon) not for approving the change. I did not have the time to take a look. Sorry for that!

Regarding the change, it seems that the two fields, `useAlertDataAsTemplate` and `useAlertDataForTemplate`, where introduced in https://github.com/elastic/kibana/pull/170162/files#diff-bab6c06ce392479bb2c15c6ea7649f51589b94dc733e112f9822d34397d942d4R68 (`x-pack/plugins/alerting/server/application/rule/schemas/action_schemas.ts`). It seems that `useAlertDataAsTemplate` is the schema for the rule client and SO and `useAlertDataForTemplate` for the request and response. The `transformRawActionsToDomainActions` in my PR followed the schema already defined in previous PRs. I agree that we should have only one and `useAlertDataAsTemplate` is the wrong one as the request schema accepts only `useAlertDataForTemplate` so probably it is not possible to persist `useAlertDataAsTemplate` in ES. However, I am not sure if in previous versions of Kibana the `useAlertDataAsTemplate` slipped somehow to ES.

Reply:
Thanks Christos, I know the ack was for the message :) This change was to make the CI green, wasn't going to merge it without your approval.